fdbclient/NativeAPI.actor.h

```diff
@@ -132,19 +132,24 @@ struct TransactionOptions {
 	bool readOnly : 1;
 	bool firstInBatch : 1;
 	bool includePort : 1;
+	bool reportConflictingKeys : 1;
 
 	TransactionOptions(Database const& cx);
 	TransactionOptions();
 
 	void reset(Database const& cx);
 };
 
+class ReadYourWritesTransaction; // workaround cyclic dependency
 struct TransactionInfo {
 	Optional<UID> debugID;
 	TaskPriority taskID;
 	bool useProvisionalProxies;
+	// Used to save conflicting keys if FDBTransactionOptions::REPORT_CONFLICTING_KEYS is enabled
+	// shared_ptr used here since TransactionInfo is sometimes copied as function parameters.
+	std::shared_ptr<ReadYourWritesTransaction> conflictingKeysRYW;
```

Comment:
```suggestion
	Reference<ReadYourWritesTransaction> conflictingKeysRYW;
```
Reference should be faster than shared_ptr

Reply:
As we discussed, `Reference` is better to be used here. 
But one problem is that `Reference` needs the underlying type to have `delref` function. 
However, since we use forward declaration here to avoid cyclic dependency.
It does not compile since the compiler knows nothing about `ReadYourWritesTransaction` here.

Reply:
Discussed in person. Keep it as `share_ptr` for now. May solve the issue in the future.

Reply:
Could this be solved by forward declaring `TransactionInfo` here and defining it the cpp file? I'm not sure it's used anywhere else.

Reply:
I think that would be one option. But we decided to solve the whole problem differently which will get rid of this member altogether.